samcli/local/lambdafn/zip.py

```diff
@@ -77,7 +138,7 @@ def _set_permissions(zip_file_info, extracted_path):
     """
 
     # Permission information is stored in first two bytes.
-    permission = zip_file_info.external_attr >> 16
+    permission = (zip_file_info.external_attr >> 16) & 511
```

Comment:
Can we consider adding a comment explaining the `& 511` part especially? May be useful to note for posterity/later development.

Reply:
Since @bubba-h57 wrote the initial pull request he might be able to give some background.